edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent.Test/MerticsSerializerTests.cs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Runtime.InteropServices;
+    using System.Text;
+    using Microsoft.Azure.Devices.Edge.Util.Test.Common;
+    using Xunit;
+
+    public class MerticsSerializerTests
+    {
+        readonly Random rand = new Random();
+
+        [Fact]
+        public void SingleMetricSerializes()
+        {
+            Metric testMetric = new Metric(DateTime.UtcNow, this.metrics[0].name, this.rand.NextDouble() * 50, this.metrics[0].tags);
+
+            byte[] data = RawMetric.MetricsToBytes(new Metric[] { testMetric }).ToArray();
+            var reconstructedValues = RawMetric.BytesToMetrics(data).ToArray();
+
+            Assert.Single(reconstructedValues);
+            TestUtilities.ReflectionEqual(testMetric, reconstructedValues[0]);
```

Comment:
you should implement IEquatable for Metric type and don't need to have "ReflectionEqual" method in TestUtilities.

Reply:
You can use 2 methods to test each other; it is possible that they both miss to serialize/deserialize a field from Metric.  Please split this test into 2.

Reply:
I would rather not implement a trait only for testing. It adds complexity to production code that isn't needed. ReflectionEqual should only be used by tests to validate expected values.

Reply:
I assert against the original test data, so if either serialize or deserialize drops a value the test will fail. I can't test serialize and deserialize separately, since I have no way of generating the expected binary array except for the function I'm testing.

Reply:
implemented iequatable